codersdk/workspaceagentconn.go

```diff
@@ -193,13 +193,32 @@ func (c *WorkspaceAgentConn) Close() error {
 	return c.Conn.Close()
 }
 
+type ReconnectingPTYBackendType string
+
+const (
+	ReconnectingPTYBackendTypeAuto     ReconnectingPTYBackendType = "auto"
+	ReconnectingPTYBackendTypeScreen   ReconnectingPTYBackendType = "screen"
+	ReconnectingPTYBackendTypeBuffered ReconnectingPTYBackendType = "buffered"
+)
+
+func (s ReconnectingPTYBackendType) Valid() bool {
+	switch s {
+	case ReconnectingPTYBackendTypeAuto, ReconnectingPTYBackendTypeBuffered,
+		ReconnectingPTYBackendTypeScreen:
+		return true
+	default:
+		return false
+	}
+}
+
 // WorkspaceAgentReconnectingPTYInit initializes a new reconnecting PTY session.
 // @typescript-ignore WorkspaceAgentReconnectingPTYInit
 type WorkspaceAgentReconnectingPTYInit struct {
-	ID      uuid.UUID
-	Height  uint16
-	Width   uint16
-	Command string
+	ID          uuid.UUID
+	Height      uint16
+	Width       uint16
+	Command     string
+	BackendType ReconnectingPTYBackendType
```

Comment:
When I added this I had the same thought.  It does fall back if you set nothing (or set `auto`) but the main reason I added it was for tests.

It might be enough that we test both backends as a consequence of running tests on multiple platforms, but it feels a bit sketchy to me.

An idea I had was to test the reconnecting pty directly rather than through the SDK as we currently do, but I vaguely remember hearing that pattern was discouraged (it is possible I have misremembered something).

Reply:
Is it possible to manipulate the test environments so that `screen` is available in one test case, but not available in another?  That would also allow us to test the code that falls back to the circular buffer.

Reply:
I had a somewhat janky idea to symlink bash into a directory and set `PATH` to that so screen is not available which does work.  I will push that up for now, but maybe there is a more clever way of doing this.